pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/CachedPulsarClient.java

```diff
@@ -80,7 +81,12 @@ private static PulsarClientImpl createPulsarClient(
             ClientConfigurationData clientConfig) throws PulsarClientException {
         PulsarClientImpl client;
         try {
-            client = new PulsarClientImpl(clientConfig);
+            // when `authentication` is null, the `getAuthentication` returns a new instance of `AuthenticationDisabled`.
+            // To hit the guava cache, we should set the authentication to the static instance.
+            // The `authentication` will be reconstructed by `authParams` when instantiating pulsar client.
+            clientConfig.setAuthentication(AuthenticationDisabled.INSTANCE);
+            // Use cloned config, as it will be modified after client was instantiated.
+            client = new PulsarClientImpl(clientConfig.clone());
```

Comment:
I think here directly `setAuthentication` is a very dangerous operation, because we do not guarantee that the logic in ClientConfigurationData off to get the `Authentication` implementation can not be modified. Another one is that I have submitted an issue[apache/pulsar#8980] in the Pulsar community to fix equals.

Reply:
Suggest adding a check here.

Reply:
Agree with you and wait your fix in pulsar.